sympy/sets/sets.py

```diff
@@ -1645,6 +1657,9 @@ def _contains(self, other):
     def as_relational(self, symbol):
         return false
 
+    def _eval_evalf(self, prec):
+        return self
```

Comment:
Is it necessary to provide this implementation if the superclass has not changed this method from the one inherited from `EvalfMixin`?

Reply:
I think it is nessesary because the default `_eval_evalf` returns `None`. Some classes (like `Union`) incorrectly use `_eval_evalf` on its arguments when it should use `_evalf`. The former returns `None` by default while the latter returns `self`.

Maybe I should have changed `Union` instead of `EmptySet`.